lib/src/data/repositories/card_repository.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:convert';
+
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:rescado/src/constants/rescado_constants.dart';
+import 'package:rescado/src/data/models/animal.dart';
+import 'package:rescado/src/data/models/card_action.dart';
+import 'package:rescado/src/data/models/card_filter.dart';
+import 'package:rescado/src/data/models/like.dart';
+import 'package:rescado/src/services/providers/api_client.dart';
+import 'package:rescado/src/utils/logger.dart';
+
+final cardRepositoryProvider = Provider<CardRepository>(
+  (ref) => ApiCardRepository(ref.read),
+);
+
+// All API endpoints regarding cards.
+abstract class CardRepository {
+  Future<List<Animal>> generate({required CardFilter cardFilter});
+
+  Future<List<Like>> getLiked();
+
+  Future<CardAction> addLiked({required List<Animal> animals});
+
+  Future<CardAction> deleteLiked({required List<Animal> animals});
+
+  Future<CardAction> addSkipped({required List<Animal> animals});
+
+  Future<void> deleteSkipped();
+}
+
+class ApiCardRepository implements CardRepository {
+  static final _logger = addLogger('ApiCardRepository');
+
+  final Reader _read;
+
+  ApiCardRepository(this._read);
+
+  @override
+  Future<List<Animal>> generate({required CardFilter cardFilter}) async {
+    _logger.d('generate()');
+
+    final endpoint = Uri.parse('${RescadoConstants.api}/cards/generate');
+
+    final response = List<Map<String, dynamic>>.from(await _read(apiClientProvider).postJson(
+      endpoint,
+      body: cardFilter.toJson(),
+    ) as List);
+
+    return response.map((animal) => Animal.fromJson(animal)).toList();
+  }
+
+  // region liked
+
+  @override
+  Future<List<Like>> getLiked() async {
+    _logger.d('getLiked()');
+
+    final endpoint = Uri.parse('${RescadoConstants.api}/cards/liked?detailed=true');
+
+    final response = await _read(apiClientProvider).getJson(
+      endpoint,
+    ) as List<Map<String, dynamic>>; // TODO Test/try because pretty sure this gives casting errors
```

Comment:
Should be something like
```
    final response = await _read(apiClientProvider).getJson(
      endpoint,
    ) as List<dynamic>;

    return response.map((dynamic like) => Like.fromJson(like as Map<String, dynamic>)).toList();
```

Reply:
Ok, but is that a blocker for this PR? Function is not used in this version of the codebase, so `develop` stays stable after merging, and you can apply the fix when you pull `develop` into `work/profile-view`?

I don't want to write code that is already written. 😅